client/ChatterApp.jsx

```diff
@@ -0,0 +1,105 @@
+ChatterApp = React.createClass({
+  mixins: [ReactMeteorData],
+
+  getInitialState: function() {
+     return {
+              chatState: "open",
+              room: null,
+              header: "Channels",
+              view: "roomList",
+              joinedRooms: [],
+              otherRooms: []
+            };
+   },
+
+  getMeteorData () {
+    const roomsHandle = Meteor.subscribe("chatterRooms");
+    const userRoomsHandle = Meteor.subscribe("chatterUserRooms");
+    const subsReady = roomsHandle.ready() && userRoomsHandle.ready();
+
+    let joinedRooms = [];
+    let otherRooms = [];
+
+    if (subsReady) {
+      var userRooms = Chatter.UserRoom.find({"userId": Meteor.userId()});
+      var roomIds = userRooms.map(function(userRoom) { return userRoom.roomId });
+      joinedRooms = Chatter.Room.find({"_id": {$in:roomIds}}).fetch();
+      otherRooms = Chatter.Room.find({"_id": {$nin:roomIds}}).fetch();
+    }
+
+    return {
+      joinedRooms: joinedRooms,
+      otherRooms: otherRooms,
+      subsReady: subsReady
+    }
+  },
+
+  goToRoom(roomId, roomName) {
+    this.setState({
+      room: roomId,
```

Comment:
"room" og "roomId" er to forskjellige ting. room virker som et objekt som holder info, mens roomId er en string. Du sparer deg selv mye forvirring med å være konsekvendt på dette, så bedre med roomId